pkg/services/preference/model.go

```diff
@@ -94,6 +96,29 @@ func (j *PreferenceJSONData) FromDB(data []byte) error {
 	return dec.Decode(j)
 }
 
+func (j *PreferenceJSONData) Scan(val interface{}) error {
+	switch v := val.(type) {
+	case []byte:
+		if len(v) == 0 {
+			return nil
+		}
+		err := json.Unmarshal(v, &j)
```

Comment:
Since there is nothing after the switch statement, can drop the following line and `return json.Unmarshal([]byte(v), &j)` (and same below in the `string` case).